src/xpk/core/gcluster_manager.py

```diff
@@ -44,8 +45,10 @@ class GclusterManager:
   def __init__(
       self,
       gcluster_command_runner: CommandRunner,
+      remote_state_client: Optional[RemoteStateClient | None],
```

Comment:
`Optional[X] = X | None` ([docs](https://docs.python.org/3/library/typing.html#typing.Optional)), so here either `Optional` or `| None` is redundant.